test/ngMessages/messagesSpec.js

```diff
@@ -485,6 +485,51 @@ describe('ngMessages', function() {
     });
   });
 
+
+  it('should clean-up the ngMessage scope when a message is removed',
+    inject(function($compile, $rootScope) {
+      function countWatchers() {
```

Comment:
There is `$rootScope.$countWatchers()` (see https://docs.angularjs.org/api/ngMock/type/$rootScope.Scope).